### PR TITLE
Bulk Data API

### PR DESCRIPTION
**Pull request by ecnerwalawrence**
Bulk Data API

### CODE REVIEW

**Comment by honteng** [comment]
I guess @takeshinoda wants to use this?

**Comment by takeshinoda** [comment]
Thank you for mention. Oh~~~ I want to use this code. But, FACloud release schedule...

Will you more commit for this API? @ecnerwalawrence

File: core.go
Comment (ethankan) [inline core.go:152]: This structure is specific to time-series data. Instead, we want to keep the data as opaque blob. Just like "event" is basically eventType and opaque eventData, the bulk data will be "streamID" and an opaque data blob ([]byte).

The encoding of time-series data (into events and bulk data) should be outside of the core device-sdk package (but perhaps as a sub-package) since not all clients are using TSDB.

Comment (ecnerwalawrence) [reply core.go:152]: done. please review.

**Comment by ecnerwalawrence** [comment]
@takeshinoda I will merge this as soon as possible. i'm waiting on @ethankan approval.

File: session.go
Comment (takeshinoda) [inline session.go]: This comment is `SendEvent` function?

File: session_test.go
Comment (takeshinoda) [inline session_test.go]: This method is not need exported names.
```suggestion
func readStream(conn net.Conn) bool {
```

File: session_test.go
Comment (takeshinoda) [inline session_test.go]: ditto

File: session_test.go
Comment (takeshinoda) [reply session_test.go]: Useful test helper function... nice.

**Review by ethankan — APPROVED** [review]
Looks good. thanks.

File: session_test.go
Comment (ecnerwalawrence) [reply session_test.go]: it is a test, but sure.

**Comment by ecnerwalawrence** [comment]
@takeshinoda I made changes based on your comments.

**Pull request merged**
